packages/nx/src/utils/nx-plugin.ts

```diff
@@ -197,17 +199,27 @@ export function registerPluginTSTranspiler() {
   if (!tsNodeAndPathsRegistered) {
     // nx-ignore-next-line
     const ts: typeof import('typescript') = require('typescript');
+    const tsConfigName = existsSync('tsconfig.base.json')
+      ? 'tsconfig.base.json'
+      : existsSync('tsconfig.json')
+      ? 'tsconfig.json'
+      : null;
+    const tsConfig: Partial<ts.ParsedCommandLine> = tsConfigName
+      ? readTsConfig(tsConfigName)
+      : {};
 
     registerTsConfigPaths(join(workspaceRoot, 'tsconfig.base.json'));
     registerTranspiler({
       lib: ['es2021'],
       module: ts.ModuleKind.CommonJS,
       target: ts.ScriptTarget.ES2021,
       inlineSourceMap: true,
-      esModuleInterop: true,
+      esModuleInterop: tsConfig.options.esModuleInterop ?? false,
```

Comment:
This would break if there is no `tsConfigName` and `tsConfig` gets value from line 209 `{}`

Reply:
The same goes for the following `tsConfig.options.*` checks

Reply:
Yes, it should take advantage of nullish coalescence. Thanks, will fix.